### PR TITLE
[MP-7746] feat: Toast Component

### PR DESCRIPTION
**Pull request by sathishkumar-thangavel**
Toast notifications are used to show success, error, warning notifications. 
Stack toast not implemented in this phase

re #111

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?

This has been testes locally with storybook

### CODE REVIEW

**Comment by sathishkumar-thangavel** [comment]
@menakallg Kindly review the content

File: src/components/icon/icon.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class Icon {
   @State() svgHTML = '';
 
   private async getSVGHTML(iconName: string) {
-    const response = await fetch(getAssetPath(`assets/icons/${iconName}.svg`));
+    const response = await fetch(getAssetPath(`icon-assets/icons/${iconName}.svg`));
```
Comment (asifahmedfw) [inline src/components/icon/icon.tsx:28]: Why the rename?

File: src/components/toast/toast.scss [hunk 2]
```diff
@@ -0,0 +1,109 @@
+:host {
```
Comment (asifahmedfw) [inline src/components/toast/toast.scss:1]: Add the freshworks theme scss.

Also, all colours and font weight should be used from the theme please.

File: src/components/toast/toast.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  float: left;
+}
+
+.cross {
+  float: right;
+  cursor: pointer;
+  margin-right: 6px;
+}
+
+@keyframes fadeOutRight {
```
Comment (asifahmedfw) [inline src/components/toast/toast.scss:78]: The animation should be the same for all. DSM doesn't have anything else. Also, can we do entry animations? Refer to product implementations.

File: src/components/button/button.tsx [hunk 4]
```diff
@@ -36,6 +36,11 @@ export class Button {
    */
   @Prop() modalTriggerId = '';
 
+  /**
+   *  Accepts the id of the fw-toast component to open it on click
+   */
+  @Prop() toastTriggerId = '';
```
Comment (asifahmedfw) [inline src/components/button/button.tsx:42]: Do we need this? Generally, modals have button use cases, toasts generally don't.

**Review by asifahmedfw — CHANGES_REQUESTED** [review]

File: src/components/icon/icon.tsx (hunk 1 above)
Comment (sathishkumar-thangavel) [reply src/components/icon/icon.tsx:28]: there's an existing issue that we can't use 'assets' directory from more than one component. This is the suggested solution to the issue

File: src/components/toast/toast.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+@Component({
+  tag: 'fw-toast',
+  styleUrl: 'toast.scss',
+  assetsDirs: ['toast-assets'],
+  shadow: true,
+})
+export class Toast {
+
+  @State() open = false;
```
Comment (shravan-balasubramanian) [inline src/components/toast/toast.tsx:11]: Can you add comments for the states and props for it to reflect in the docs?

File: src/components/toast/toast.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+  @State() iconName: string;
+
+  @State() isMouseHover: boolean;
```
Comment (shravan-balasubramanian) [inline src/components/toast/toast.tsx:15]:
```suggestion
  @State() isMouseHovered: boolean;
```

File: src/components/toast/toast.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  @State() svgHTML = '';
+
+  @State() isTimeOut = false;
```
Comment (shravan-balasubramanian) [inline src/components/toast/toast.tsx:19]:
```suggestion
  @State() isTimedOut = false;
```

File: src/components/toast/toast.tsx [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+  render() {
+    return (
+      <Host
+        class={`toast ${this.position} ${this.type} ${this.open ? 'is-open' : ''} ${this.fadeOut ? 'fade-out' : ''}`}
+        aria-hidden={this.open ? 'false' : 'true'}
+        onmouseover={() => this.mouseHover(true)}
+        onmouseout={() => this.mouseHover(false)}
+        >
+        {/* <div class={`toast ${this.position} ${this.type}`}> */}
```
Comment (shravan-balasubramanian) [inline src/components/toast/toast.tsx:134]: Do we need this?

File: src/components.d.ts [hunk 9]
```diff
@@ -473,6 +473,37 @@ export namespace Components {
...
+        "sticky": boolean;
+        /**
+          * Time duration of the toast visibility
+         */
+        "timeout": number;
+        "trigger": (configs: object) => Promise<void>;
+        /**
+          * Type of the toast - success,failure, warning, inprogress
+         */
+        "type": "success" | "error" | "warning" | "inprogress";
```
Comment (shravan-balasubramanian) [inline src/components.d.ts:505]:
```suggestion
        "type": "success" | "failure" | "warning" | "inprogress";
```
Just a suggestion 😄

File: src/components/toast/toast.scss [hunk 10]
```diff
@@ -0,0 +1,100 @@
+@import "../../styles/freshworks.scss";
+
+:host {
+  display: none;
```
Comment (shravan-balasubramanian) [inline src/components/toast/toast.scss:4]: Can you add font family here?

File: src/components.d.ts (hunk 9 above)
Comment (sathishkumar-thangavel) [reply src/components.d.ts:505]: This not just for failure alone but can be used for any errors.

File: src/components/toast/toast.tsx (hunk 8 above)
Comment (sathishkumar-thangavel) [reply src/components/toast/toast.tsx:134]: forgot to remove it after checked. will remove

File: src/components/toast/toast.scss (hunk 10 above)
Comment (sathishkumar-thangavel) [reply src/components/toast/toast.scss:4]: it's added in freshworks.scss file that imported above

**Review by shravan-balasubramanian — APPROVED** [review]

**Review by asifahmedfw — APPROVED** [review]

**Pull request merged**